### PR TITLE
[Ready] Splits the emag into two items, effectively nerfing the emag.

### PR DESCRIPTION
**Pull request by Firecage**
This splits the emag into two different items, the Cryptographic Sequencer and the Electromagnetic Sequencer. The Cmag and Emag.

The Cryptographic Sequencer is used to unlock shit. While the Electromagnetic Sequencer is used for the whacky shit like emagging vapes and borgs. Both costs 4 TC's in the uplink, for a total cost of 8 TC.

Can't find any problems while testing. Though I am unsure if I may have missed one or two things. But if not, nothing seems to have horribly broken for once.

:cl: Firecage
balance: NanoTrasen Bureau of Intelligence Notice: It appears the Syndicate Command has decided to split the technology of an item commonly known as the Emag into two separate items to both save manufacturing costs, and to try to reduce operatives who turn rogue due to power. Be on the watch for a suspicious, black card.
/:cl:

### CODE REVIEW

**Comment by praisenarsie** [comment]
my favourite pr

**Comment by CPTANT** [comment]
Is there a list of what this leaves for the cryptographic sequencer?

File: code/modules/mining/abandoned_crates.dm [hunk 1]
```diff
@@ -188,7 +188,7 @@
 
 /obj/structure/closet/crate/secure/loot/attackby(obj/item/W, mob/user)
 	if(locked)
-		if(istype(W, /obj/item/card/emag))
+		if(istype(W, /obj/item/card/lockhacker))
 			boom(user)
```
Comment (Cyberboss) [inline code/modules/mining/abandoned_crates.dm]: This should be in the _act proc

**Review by Cyberboss — CHANGES_REQUESTED** [review]

File: code/modules/mob/living/simple_animal/bot/bot.dm [hunk 2]
```diff
@@ -37,7 +37,7 @@
 	var/bot_name
 
 	var/list/player_access = list() //Additonal access the bots gets when player controlled
-	var/emagged = FALSE
+	var/emagged = FALSE  //if 1, then a lockhacker was used. If 2, then an emag was used.
```
Comment (JJRcop) [inline code/modules/mob/living/simple_animal/bot/bot.dm]: Make these defines in [code/__DEFINES/misc.dm](https://github.com/tgstation/tgstation/blob/511e357472fe75be261f904bce00c215ea71ed76/code/__DEFINES/misc.dm), e.g.
```DM
#define EMAGGED_LOCKHACK 1
#define EMAGGED_EMAG 2
```
then usage would be as so: ```if(emagged == EMAGGED_EMAG)``` or ```if(emagged == EMAGGED_LOCKHACK)```

Comment (Firecage) [reply code/modules/mob/living/simple_animal/bot/bot.dm]: Will be changing those tomorrow.

**Comment by pubby** [comment]
>The Cryptographic Sequencer does all the fun and whacky stuff. While the Electromagnetic Sequencer is used to unlock shit.

That's the opposite of what I expected when I read the names. Breaking crypto implies breaking locks.

File: code/modules/mob/living/simple_animal/bot/bot.dm (hunk 2 above)
Comment (Firecage) [reply code/modules/mob/living/simple_animal/bot/bot.dm]: Wish there was a way for me to NOT make this one outdated, since I still have to do this.

**Comment by Firecage** [comment]
Pubby, read the new description.

**Comment by DoctoreMad** [comment]
**I called this shit 1 year ago**.
**I can't prove it but I did**. **Believe in me who believes in you**.

**Comment by DoctoreMad** [comment]
>Insanus Sequencer

Okay that's a pretty bad name.
E-mag should be the one to keep its name if anything.

**Comment by Firecage** [comment]
Insanus is a latin translation of insane, which can also mean loko, crazy, etc. Since the Insanus Sequencer does all the whacky shit. Anyway, not changing names again.

**Comment by AnonymousNow** [comment]
If you have to have this, make the Cryptographic Sequencer the lock-breaker, and the Protocol Scrambler the fun stuff doer.

They'll both be under the banner of "Electromagnetic Card", which should exist as the combined legacy item and cost marginally more telecrystals than the other two combined.

**Comment by DoctoreMad** [comment]
I'm not sure if you understand, e-mag's name is iconic at this point.

**Comment by Firecage** [comment]
Anon, refresh and check the description. Already changed so Cryptographic sequencer is the lock breaker. And there is a reason why I made the other one into a card as well.

**Comment by Ergovisavi** [comment]
This is a stupid idea.

Is the emag too powerful for its cost? Yes, absolutely.

Is that actually a good thing? Yes. Because it incentivizes traitors _doing shit_ instead of being complete pussies and sitting around for fifty minutes. You want to incentivize antagonists actually going out and doing their jobs, that is, sabotaging and terrorizing the station, and if that means an item is a little too good for its cost, so be it. The alternative is significantly worse, a cost based system where everyone is too chickenshit to spend points and go out and wreck the place.

The emag gets people into places to do stuff. Harming that is harming the game, even if from pure spreadsheet design, it's out of line.

**Comment by Y0SH1M4S73R** [comment]
I honestly had a more interesting name for the "fuck shit up" device for my own idea of splitting the emag in two. While I'm fine with the ID-bypassing device being called "Cryptographic Sequencer", I was thinking of calling the shit-fucker "Subroutine Inscriber", a name based in my interpretation that the shit-fucking function of the emag injects function-altering code into the hacked object.

**Comment by Firecage** [comment]
Ergo. And to that I will say this. If they just need to get INTO places? Then they can do that easier now, they can just buy the Cryptographic sequences for 5 TC's instead of 6 TC's. The Crypto Sequencer opens doors and lockers, and APC's and bot covers and shit.

The Emag as it was, was RIDICULOUSLY overpowered and overbloated with shitty features. There was 3 rational options. 1. Remove the emag. 2. Keep the emag as is, but remove a large number of functions from it. 3. Split the emag into 2 items, with both combined having all the options the emag previously had.

**Comment by DoctoreMad** [comment]
​>Insanus Sequencer
​>I really don't care about things already being iconic.

Nothing personal, then? I guess I kid.
*Teleports behind exit*
heh.......

**Comment by Iamgoofball** [comment]
it's not about iconic, it's about easily remembering the names

emag-cmag are easy to remember, electro mag, crypto mag

insaneus mag is the sort of shit you expect that is like naming a gun FK-69, edgy for the sake of edgy

**Comment by Firecage** [comment]
Well, if the biggest issue is of people complaining about the name, well, I take that as a good sign.

**Comment by Ergovisavi** [comment]
Firecage, it doesn't matter if something is overpowered, look at the effects it has on the game. The net benefit of the item is that it produces traitorous gameplay, and hindering that is an idiotic endeavor.

**Comment by Firecage** [comment]
Ergo, then you can merely buy either one or both of these items and spend the rest of the TC wisely. If you wish to have two items together which would be highly overpowered due to the immense amount of functions, then you can pay a much higher price.

**Comment by Firecage** [comment]
But fine, I will change Insanus Sequencer back to Electromagnetic Sequencer.

**Comment by Qustinnus** [comment]
I wouldn't make them the same price, I'd say the door opening card should cost more , while the card that is only useful on specific machines should cost less.

**Comment by Bluespace** [comment]
RIP the best traitor item.
I'll m-miss you.

**Comment by AnonymousNow** [comment]
I don't know. I can't help but get the feeling that this is a reactive change, jumped into with both feet at speed.

Who hurt you, man?

**Comment by Firecage** [comment]
Perhaps you shouldn't make accusative statements without anything to back it up? Since this is looking at a very OP item, which I frequently use myself. And seeing that it either has to go, get a severe nerf, or be split into more than one item due to over overpowered it is.

**Comment by Firecage** [comment]
@DoctoreMad Happy now?

**Comment by DoctoreMad** [comment]
>I wouldn't make them the same price, I'd say the door opening card should cost more , while the card that is **only useful on specific machines** should cost less.

Who the heck is behind this post, Todd Howard

**Comment by Firecage** [comment]
Possibly. (Who is Todd Howard?)

**Comment by Repukan** [comment]
Buy my game

**Comment by DoctoreMad** [comment]
Todd Howard is a good guy, and he is able to expertly talk his way into many things since he has a long reputation for being trustworthy. He is very good at chosen peopling others.

**Comment by ChangelingRain** [comment]
the one that unlocks stuff should cost 4, and the one that messes with stuff should cost either 3 or 4, since the current cost is 6.

**Comment by Firecage** [comment]
@ChangelingRain Sure, will look for better price rebalancing when I do another pr update tomorrow, which will include a few other things to.

**Comment by optimumtact** [comment]
Excellent PR

**Comment by Supermichael777** [comment]
Rename them to be less similar.
Have one be "Illegal firmware card" and leave the other as the cryptographic sequencer, this way the purpose is more clear for each device.

**Comment by kevinz000** [comment]
may i suggest that you make it still cost 6TC for both and we can rebalance them later? too many changes lead to people getting mad.

also if firecage drops this ping me and i'll continue this.

**Comment by Anonymous583** [comment]
Trash PR. An e-mag is already high cost enough, Traitors rarely pick anything else with it outside of a weapon due to how immensely good it is already. The most I ever get with an e-mag is whatever is in the discount section.

**Comment by DoctoreMad** [comment]
@kevinz000 If half a tool's functionality was cut off you'd have the decency to decrease the cost by at least 1 TC, you'd think that wouldn't be controversial.

**Comment by YakumoChen** [comment]
@DoctoreMad I think he means 'buy both for 6 TC total'. 

If you're going to separate them, balancing the cost of them together should be a separate balance issue and therefore in a different PR. We don't need to make an endless debate of balance issues when we can just tackle them one at a time.

**Comment by YakumoChen** [comment]
I also have reservations on if this is an i ded nerf as well based on the branch name alone

**Comment by Repukan** [comment]
Please, if people coding out things that killed them were grounds to reject the PR, half of kevinz code wouldn't be in.

**Comment by ShadowDeath6** [comment]
I've never been a big fan of the emag since all the powercreep it's gotten over time. Considering how potent it is at being able to open any door or locker that isn't explicitly bolted shut is extremely good and makes getting traitor objectives much easier than they should be

on top of making breaking and enteri

**Comment by FantasticFwoosh** [comment]
it looks like you prodded this PR a bit with what stuff is named etc, but in concept its good.

A basic omni-lockpick (which leaves operatives with considerably more options) and a tool akin to using a big magnet to get emag interactions. 10tc each, seems good.

- This also pushes up the rarity of existing 'full' emags

**Comment by kevinz000** [comment]
>10tc each

fwoosh sir are you ok


anyways @Firecage make them cost 6TC in total for now imo, we can go to the nitty-gritty of balance later on.

**Comment by Firecage** [comment]
@kevinz000 Mmm, I like Joans idea of 4 tc each. Or 4 for one and 3 for the other. If you can buy both for the same amount of TC, well, yah, not much point then.

**Comment by Firecage** [comment]
PS. Kinda funny when people think this PR is 'Reee, i ded, nerf plz'. When infact, I FREQUENTLY use the emag...and am kind of one of the salt creators when I emag absolutely everything in sight.

**Comment by CPTANT** [comment]
What emag functions that are not unlocking things are there currently in the game anyway?

I know:

hacking borgs
hacking comms
shuttle emergency launch
null crates
disabling holo safety
disabling sleeper safety
fucking up the cloner
turning safety pins into regular pins
hacking medi/floor/security bots

**Comment by KorPhaeron** [comment]
This definitely needs some period of testmerging. Looked at it on its own the emag is a silly, feature creeped item but is making traitor more difficult to play for med doctors/librarians/new people etc really going to improve rounds?

**Comment by KorPhaeron** [comment]
Also you should switch the names.

**Comment by Arianya** [comment]
@KorPhaeron This overall lowers the cost of the aspects of the emag that those roles need, access.

**Comment by Firecage** [comment]
@KorPhaeron Definitely not switching names a third time. Though currently Cryptographic is for unlocking stuff, which makes sense as that is what Crypto implies. Electromagnetic is for messing stuff up...so kinda like a very specialized version of an EM pulse. So that kinda makes sense to.

**Comment by KorPhaeron** [comment]
EM pulses in our game destroy things, they don't magically reprogram things

Crypto Sequencer sounds much closer to hacking gibberish

**Comment by CPTANT** [comment]
Ok what if they don't both get "sequencer" in there name?

Call one the "Syndicate machine instruction card" or something.

**Comment by CPTANT** [comment]
I still think a list of actual emag functions would be good before making a decision on this

**Comment by Arianya** [comment]
@KorPhaeron What about a Ionic Sequencer

Ion has a already established purpose for doing silly shit.

**Comment by Firecage** [comment]
@KorPhaeron No, seriously though. It's a pain to change everything. Since for consistency, I also change the proc names and such....so it is a bloody pain to change the names and I have already done it twice. Besides. Emag and Cmag. I kind of like that.

**Comment by nicbn** [comment]
What if you name the crypto sequencer keygen sequencer? keygen or keymag sounds way better than cmag.

**Comment by nerumeru** [comment]
could consider a 1 tc kit to merge the two sequencers back into one item if this goes through (and if that specific nerf's not the main goal of the pr anyway)

ultimately a 50% price increase at that point (and more 1 tc options! wao)

**Comment by KorPhaeron** [comment]
@firecage cant you just mass find/replace

**Comment by Firecage** [comment]
Kor...god no. Do you WANT me to break everything? Now, if I was JUST making a PR to rename, say, emags into something else. Then sure, that would have worked perfectly with no problems(maybe...I hope).

But, doing that for any other PR is bound to just break everything and it will even be more annoying to fix it. Besides, no one can agree on the names, since I already had to do name changes twice. How about we just see how these names fits for now?

**Comment by DoctoreMad** [comment]
>Definitely not switching names a third time

Just give branch access to someone to knows how to use queries to replace names quickly.

**Comment by JJRcop** [comment]
You can find/replace your diff

**Comment by nicbn** [comment]
Pic needed.

**Comment by Ratvar** [comment]
Allow to combine both to get classical emag for the convinience. Cause why not.

**Comment by Firecage** [comment]
No Ratvar, not doing that.

**Comment by Firecage** [comment]
Ditto with the other PR. Not much point in doing PR's which requires changelogs if only serious changelogs 
 and boring monotone changelogs are allowed now

**Pull request closed without merging**